src/osparc_python_runner/main.py

```diff
@@ -91,40 +54,63 @@ def ensure_requirements(code_dir: Path) -> Path:
         # deduce requirements using pipreqs
         logger.info("Not found. Recreating requirements ...")
         requirements = code_dir / "requirements.txt"
-        run_cmd(f"pipreqs --savepath={requirements} --force {code_dir}")
+        subprocess.run(
+            f"pipreqs --savepath={requirements} --force {code_dir}".split(),
+            shell=False,
+            check=True,
+            cwd=INPUT_FOLDER,
+        )
 
         # TODO log subprocess.run
 
     else:
         requirements = requirements[0]
+        logger.info(f"Found: {requirements}")
     return requirements
 
 
-def setup():
-    logger.info("Cleaning output ...")
-    clean_dir(output_dir)
-
-    # TODO: snapshot_before = list(input_dir.rglob("*"))
-
-    # NOTE The inputs defined in ${INPUT_FOLDER}/inputs.json are available as env variables by their key in capital letters
-    # For example: input_1 -> $INPUT_1
-    #
-
-    logger.info("Processing input ...")
-    unzip_dir(input_dir)
+# TODO: Next version of integration will take care of this and maybe the ENVs as well
+def _ensure_output_subfolders_exist() -> Dict[str, str]:
+    output_envs = {}
+    for n in range(1, NUM_OUTPUTS + 1):
+        output_sub_folder_env = f"OUTPUT_{n}"
+        output_sub_folder = OUTPUT_FOLDER / OUTPUT_SUBFOLDER_TEMPLATE.format(n)
+        # NOTE: exist_ok for forward compatibility in case they are already created
+        output_sub_folder.mkdir(parents=True, exist_ok=True)
+        output_envs[output_sub_folder_env] = f"{output_sub_folder}"
+    logger.info(
+        "Output ENVs available: %s",
+        json.dumps(output_envs, indent=2),
+    )
+    return output_envs
+
+
+def _ensure_input_environment() -> Dict[str, str]:
+    input_envs = {
+        f"INPUT_{n}": os.environ[f"INPUT_{n}"] for n in range(1, NUM_INPUTS + 1)
+    }
+    logger.info(
+        "Input ENVs available: %s",
+        json.dumps(input_envs, indent=2),
+    )
+    return input_envs
 
-    # logger.info("Copying input to output ...")
-    # copy(input_dir, code_dir)
 
-    logger.info("Searching main entrypoint ...")
-    main_py = ensure_main_entrypoint(input_dir)
-    logger.info("Found %s as main entrypoint", main_py)
+def setup():
+    input_envs = _ensure_input_environment()
+    output_envs = _ensure_output_subfolders_exist()
+    logger.info("Available data:")
+    os.system("ls -tlah")
 
-    logger.info("Searching requirements ...")
-    requirements_txt = ensure_requirements(input_dir)
+    user_code_entrypoint = _find_user_code_entrypoint(INPUT_FOLDER)
+    requirements_txt = _ensure_pip_requirements(INPUT_FOLDER)
 
     logger.info("Preparing launch script ...")
     venv_dir = Path.home() / ".venv"
+    bash_input_env_export = [f"export {env}={path}" for env, path in input_envs.items()]
+    bash_output_env_export = [
+        f"export {env}={path}" for env, path in output_envs.items()
```

Comment:
WARNING: shouldn't you put quotes in the path? for example ...
```cmd
➜   export FOO=/data/a file/with spaces
export: not valid in this context: file/with
➜   export FOO="/data/a file/with spaces"
➜   echo $FOO
/data/a file/with spaces

Reply:
BTW this is what happened to us  and was fixed in this commit https://github.com/ITISFoundation/osparc-python-runner/commit/846c6dbcde7a4a29f1cd44a050a2fbac9a3d475f#diff-7120f41bf9df684771f3a411e9513094f5be0bef68c1f2ae12f682dff2302520R136

Reply:
good point